Classes/git/PBGitRepositoryWatcher.m

```diff
@@ -32,93 +51,48 @@ void PBGitRepositoryWatcherCallback(ConstFSEventStreamRef streamRef,
 									void *_eventPaths,
 									const FSEventStreamEventFlags eventFlags[],
 									const FSEventStreamEventId eventIds[]){
-	PBGitRepositoryWatcherCallbackBlock block = (__bridge PBGitRepositoryWatcherCallbackBlock)clientCallBackInfo;
-	
-	NSMutableArray *changePaths = [[NSMutableArray alloc] init];
+	PBGitRepositoryWatcher *watcher = (__bridge PBGitRepositoryWatcher *)clientCallBackInfo;
+
+	NSMutableArray *gitDirEvents = [NSMutableArray array];
+	NSMutableArray *workDirEvents = [NSMutableArray array];
 	NSArray *eventPaths = (__bridge NSArray*)_eventPaths;
 	for (int i = 0; i < numEvents; ++i) {
 		NSString *path = [eventPaths objectAtIndex:i];
 		PBGitRepositoryWatcherEventPath *ep = [[PBGitRepositoryWatcherEventPath alloc] init];
 		ep.path = [path stringByStandardizingPath];
 		ep.flag = eventFlags[i];
-		[changePaths addObject:ep];
 
+
+		if ([ep.path hasPrefix:watcher.gitDir]) {
+			// exclude all changes to .lock files
+			if ([ep.path hasSuffix:@".lock"]) {
+				continue;
+			}
+			[gitDirEvents addObject:ep];
+		} else if ([ep.path hasPrefix:watcher.workDir]) {
+			[workDirEvents addObject:ep];
+		}
 	}
-	if (block && changePaths.count) {
-		block(changePaths);
+
+	if (workDirEvents.count) {
+		[watcher handleWorkDirEventCallback:workDirEvents];
+	}
+	if (gitDirEvents.count) {
+		[watcher handleGitDirEventCallback:gitDirEvents];
 	}
 }
 
 @implementation PBGitRepositoryWatcher
 
-@synthesize repository;
+@synthesize repository, gitDir, workDir;
```

Comment:
gitDir and workDir don't need to be synthesized here (the method declaration of each is enough).